### ISSUE TITLE
[iOS] Update not applied on app restart

### ISSUE THREAD

**Issue by bonesyblue** [microsoft/react-native-code-push#2650]
### Steps to Reproduce
The following code manually checks for the availability of a CodePush update and, if available, proceeds to download and install the update.

```typescript
const update = await CodePush.checkForUpdate();

if (update) {
  const updateBundle = await update.download();
  // Update should be applied on next app restart.
  await updateBundle.install(CodePush.InstallMode.ON_NEXT_RESTART);
  await CodePush.notifyAppReady();
}
```

My root component:
```typescript
const CodePushApp = CodePush({
  checkFrequency: CodePush.CheckFrequency.MANUAL,
})(AppWrapper);

export default CodePushApp;
```

### Expected Behavior
When restarting the application, the update bundle is applied.

### Actual Behavior
The update bundle is only applied if I explicitly call `CodePush.restartApp()`. However, the update is lost if I then restart the application.

> 💡 Note: There are no issues with the same code on Android.

### Things I've Tried
I've followed the [troubleshooting guide](https://learn.microsoft.com/en-us/appcenter/distribution/codepush/rn-troubleshooting) and enabled CodePush logs for my release target. There are no issues logged and the update appears to be successful.

```
[CodePush] Applying full update.
[CodePush] Verifying hash for folder path: ...
[CodePush] Manifest string: ( ...
[CodePush] Expected hash: ..., actual hash: ...
[CodePush] The update contents succeeded the data integrity check.
[CodePush] JWT signature verification succeeded, payload content: ...
[CodePush] The update contents succeeded the code signing check.
```

I also tried replacing my manual code by just calling [CodePush.sync()](https://learn.microsoft.com/en-us/appcenter/distribution/codepush/rn-api-ref#codepushsync), however, I experience exactly the same behaviour on iOS.

### Environment

* react-native-code-push version: **7.1.0**
* react-native version: **0.70.14**
* iOS version: **16.3.1**
* Does this reproduce on a debug build or release build? **release**
* Does this reproduce on a simulator, or only on a physical device? **physical device: iPhone 12 Pro Max**

**Comment by bonesyblue** [comment]
Hi all 👋 I spent some time today examining the issue I described above. As I'm working on a production project, I created a separate fresh project for testing purposes. I'm pleased to report that I couldn't reproduce the issue described above in the new project, indicating that the problem I'm experiencing is not directly related to `react-native-code-push`.

I'll continue my investigation in the linked repository below, but for now, I'll close this issue.

https://github.com/bonesyblue/byte-sync
